packages/alchemy/src/react/hooks/useDisconnectSigner.ts

```diff
@@ -0,0 +1,43 @@
+"use client";
+
+import { useMutation, type UseMutateFunction } from "@tanstack/react-query";
+import { ClientOnlyPropertyError } from "../../config/errors.js";
+import { useAlchemyAccountContext } from "../context.js";
+import { useSigner } from "./useSigner.js";
+
+export type UseDisconnectSignerResult = {
+  disconnect: UseMutateFunction<void, Error, void, unknown>;
+  isDisconnecting: boolean;
+  error: Error | null;
+};
+
+export function useDisconnectSigner(): UseDisconnectSignerResult {
+  const { queryClient } = useAlchemyAccountContext();
+  const signer = useSigner();
+
+  const {
+    mutate: disconnect,
+    isPending: isDisconnecting,
+    error,
+  } = useMutation(
+    {
+      mutationFn: async () => {
+        if (!signer) {
+          throw new ClientOnlyPropertyError("signer");
+        }
+
+        return signer.disconnect();
+      },
```

Comment:
```suggestion
      mutationFn: signer.disconnect,
```

the client only check should be happening in the `useSigner` hook already (not in the hook itself, but actually even lower in the `getter` for it